src/zarr/codecs/pipeline.py

```diff
@@ -247,7 +248,17 @@ async def read_batch(
                 if chunk_array is not None:
                     out[out_selection] = chunk_array
                 else:
-                    out[out_selection] = chunk_spec.fill_value
+                    fill_value = chunk_spec.fill_value
+
+                    if fill_value is None:
```

Comment:
Something we may consider in the future is to parameterize the `CodecPipeline` class with the `zarr_format` of the calling Array. This would allow us feel more confident that workarounds like this one only apply to v2 data.